### PR TITLE
qmp_command:add command qom-list-types

### PR DESCRIPTION
**Pull request by yiqianwei**
list the pci,pcie and all devices

ID:1896275

Signed-off-by: Yiqian Wei <yiwei@redhat.com>

### CODE REVIEW

**Comment by yiqianwei** [comment]
test result:
(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qmp_command.qmp_qom-list-types_conventional-pci-device.q35: PASS (38.11 s)

(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qmp_command.qmp_qom-list-types_pci-express-device.q35: PASS (43.19 s)

(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qmp_command.qmp_qom-list-types_pci-device.q35: PASS (42.04 s)

**Comment by yiqianwei** [comment]
@yanan-fu, @jingzhao84 ,Could you help to review this patch ? thanks

File: qemu/tests/cfg/qmp_command.cfg [hunk 1]
```diff
@@ -193,3 +193,15 @@
         - qmp_query-machines:
             qmp_cmd = "query-machines"
             cmd_result_check = "contain"
+        - qmp_qom-list-types_conventional-pci-device:
```
Comment (yanan-fu) [inline qemu/tests/cfg/qmp_command.cfg:196]: Suggest `qmp_qom-list-types_conventional`

File: qemu/tests/cfg/qmp_command.cfg [hunk 2]
```diff
@@ -193,3 +193,15 @@
...
+            qmp_cmd = "qom-list-types implements=conventional-pci-device"
+            cmd_result_check = "contain"
+            cmd_return_value = "[{'name': 'vfio-pci'}, {'name': 'qemu-xhci'}, {'name': 'virtio-net-pci'}, {'name': 'rtl8139'}]"
+        - qmp_qom-list-types_pci-express-device:
```
Comment (yanan-fu) [inline qemu/tests/cfg/qmp_command.cfg:200]: Suggest `qmp_qom-list-types_pcie`

File: qemu/tests/cfg/qmp_command.cfg [hunk 3]
```diff
@@ -193,3 +193,15 @@
...
+            qmp_cmd = "qom-list-types implements=pci-express-device"
+            cmd_result_check = "contain"
+            cmd_return_value = "[{'name': 'vfio-pci'}, {'name': 'qemu-xhci'}, {'name': 'virtio-blk-pci'}, {'name': 'e1000e'}]"
+        - qmp_qom-list-types_pci-device:
```
Comment (yanan-fu) [inline qemu/tests/cfg/qmp_command.cfg:204]: suggest `qmp_qom-list-types_pci`

**Comment by yiqianwei** [comment]
@yanan-fu,thanks,I have taken your suggestion,.Could you help to review this patch again?

test result:
(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qmp_command.qmp_qom-list-types_conventional.q35: PASS (39.62 s)

(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qmp_command.qmp_qom-list-types_pcie.q35: PASS (42.00 s)

(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qmp_command.qmp_qom-list-types_pci.q35: PASS (41.17 s)

**Comment by jingzhao84** [comment]
Looks good on my side
ack: jinzhao@redhat.com

**Comment by yanan-fu** [comment]
Not all the devices was checked, but ack with feature owner's opinion, only choose part of those devices(different type) as an example.

Ack

**Pull request merged**